src/main/scala/org/camunda/feel/impl/builtin/ListBuiltinFunctions.scala

```diff
@@ -215,37 +215,27 @@ object ListBuiltinFunctions {
       case List(ValList(list)) => all(list)
     }, hasVarArgs = true)
 
-  private def all(xs: List[Val]): Val = xs match {
-    case Nil => ValBoolean(true)
-    case x :: xs =>
-      x match {
-        case ValBoolean(false) => ValBoolean(false)
-        case ValBoolean(true)  => all(xs)
-        case other =>
-          all(xs) match {
-            case ValBoolean(false) => ValBoolean(false)
-            case _                 => ValNull
-          }
-      }
+  private def all(items: List[Val]): Val = {
+    items.foldLeft[Val](ValBoolean(true)) {
+      case (ValBoolean(false), _)               => ValBoolean(false)
+      case (ValBoolean(true), item: ValBoolean) => item
+      case (ValNull, ValBoolean(false))         => ValBoolean(false)
```

Comment:
❌ I think L222 might be wrong, as it could turn a null case back to false, e.g. `all([true, null, false]) => false, but should be null`. Perhaps I misunderstand this though.

Reply:
This is by intention :sweat_smile: 

In FEEL, if one value is `false` then the result is `false`. The value of `null` (or any other non-boolean value) can't change the result. Only if all other values are `true` then the value of `null` matters because it could change the result to `false`.

https://camunda.github.io/feel-scala/docs/reference/language-guide/feel-boolean-expressions#conjunction--and

Reply:
Ah that's different than I expected. Interesting